model/lifecycle.go

```diff
@@ -1426,14 +1426,18 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 	taskIds := []string{}
 	for _, b := range builds {
 		// Find the set of task names that already exist for the given build
-		tasksInBuild, err := task.Find(task.ByBuildId(b.Id).WithFields(task.DisplayNameKey))
+		tasksInBuild, err := task.Find(task.ByBuildId(b.Id).WithFields(task.DisplayNameKey, task.ActivatedKey))
```

Comment:
an alternative strategy that occurs to me is to query by `BuildId` and `DisplayName: {"$in": pairs.ExecTasks.TaskNames(b.BuildVariant)}`, without a projection. This has an advantage that we wouldn't need a separate query for each individual `fullTask` later on.

Reply:
Yeah that's a good idea, although it would also have to include `pairs.DisplayTasks.TaskNames(b.BuildVariant)`. We don't have an index for both build ID and display name but build ID narrows it down enough I would imagine. Do you agree with this approach @bsamek ?

Reply:
wrote it up but I can revert it

Reply:
> it would also have to include pairs.DisplayTasks.TaskNames(b.BuildVariant)

why is this the case? (given that execution tasks can't depend on display tasks)

Reply:
Well we need to know what display tasks exist also for the latter loop, but we could do a separate query to get those. I prefer this for simplicity but I'm not sure either way.

Reply:
ah, you're right: I forgot about that.

Reply:
but that begs the question: why do we need to check for existing display tasks?

Reply:
The logic to look for existing tasks is pre-existing: we need to know what tasks already exist so we don't attempt to re-create them.

Reply:
Is there's a possibility `pairs ` contains tasks other than 1) new tasks and 2) their dependencies?
If so, I don't have an answer for [Brian's question](https://github.com/evergreen-ci/evergreen/pull/3896#discussion_r472171335).

Reply:
Pairs contains only tasks that we want activated or created.

Reply:
I'm not sure what the question really is here.

Reply:
sorry. What I meant was to me that to me 

> Pairs contains only tasks that we want activated or created

implies we don't need to check pairs for existing display tasks.

Reply:
Oh I see. That's possible, but I'm hesitant to stop handling display tasks in this ticket in case there was a reason to handle it to begin with, and it seems out of scope of a bug fix ticket.